scopes/react/react/react.docs.md

```diff
@@ -0,0 +1,422 @@
+
+For example:
+
+```ts
+// ...
+
+const webpackConfig = require('./webpack/webpack.config');
+
+export class ReactExtension {
+
+// ...
+
+  static async provider([envs, react]: [EnvsMain, ReactMain]) {
+    const newReactEnv = react.compose([
+      react.overridePreviewConfig(webpackConfig)
+    ]);
+
+
+}
+
+// ...
+```
+
+#### overrideDevServerConfig
+
+```ts
+overrideDevServerConfig(config: Configuration): EnvTransformer
+```
+
+Overrides the Webpack configurations for the [DevServer](/docs/environments/environment-services#devserver) environment service, with a new ([webpack.config.js](https://webpack.js.org/configuration/)) configuration file.
+
+For example:
+
+```ts
+// ...
+
+const webpackConfig = require('./webpack/webpack.config');
+
+export class ReactExtension {
+
+// ...
+
+  static async provider([envs, react]: [EnvsMain, ReactMain]) {
+    const newReactEnv = react.compose([
+      react.overrideDevServerConfig(webpackConfig)
+    ]);
+
+
+}
+
+// ...
+```
+
+#### overrideJestConfig
+
+```ts
+overrideJestConfig(jestConfigPath: string): EnvTransformer
+```
+
+This method receives a path (as a string) to a configuration file . Overrides the default configurations for the Jest test runner with a new ([jest.config](https://jestjs.io/docs/en/configuration)) configuration file. This is done by passing the _path_ to the file as an argument.
+
+For example:
+
+```ts
+// ...
+
+export class ReactExtension {
+
+// ...
+
+  static async provider([envs, react]: [EnvsMain, ReactMain]) {
+    const newReactEnv = react.compose([
+      react.overrideJestConfig(require.resolve('./jest/jest.config'))
+    ]);
+
+
+}
+
+// ...
+```
+
+#### overrideBuildPipe
+
+```ts
+overrideBuildPipe(tasks: BuildTask[]): EnvTransformer
+```
+
+This method receives an array of Bit tasks. It overrides the build pipeline of a component (initiated either on a `bbit tag` or `bbit build` command). To create your own Build Task, [see here](/docs/build-pipeline/create-build-task).
+
+For example:
+
+```ts
+// ...
+
+// Import the task
+import { CustomTask } from './custom.task';
+
+export class CustomReact {
+  // ...
+
+  static async provider([envs, react]: [EnvsMain, ReactMain]) {
+    // Get the environment's default build pipeline using the 'getBuildPipe' service handler
+    const reactPipe = react.env.getBuildPipe();
+
+    // Add the custom task to the end of the build tasks sequence.
+    const tasks = [...reactPipe, new CustomTask()];
+
+    const newReactEnv = react.compose([react.overrideBuildPipe(tasks)]);
+
+    // ...
+  }
+}
+```
+
+#### overrideDependencies
+
+```ts
+overrideDependencies(dependencyPolicy: DependenciesPolicy): EnvTransformer
+```
+
+This method receives a Bit dependency-policy object. It overrides the default dependency policy for components using this environment.
```

Comment:
mention it will be merged with the react default config

Reply:
Done.